src/Instrumentation/Psr16/src/Psr16Instrumentation.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Psr16;
+
+use Composer\InstalledVersions;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanBuilderInterface;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\SimpleCache\CacheInterface;
+use Throwable;
+
+/**
+ * @psalm-suppress ArgumentTypeCoercion
+ */
+class Psr16Instrumentation
+{
+    public const NAME = 'psr16';
+
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.psr16',
+            InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-psr16')
+        );
+
+        hook(
```

Comment:
There don't seem to be any differences between these 8 hooks besides the method name. Perhaps you can extract it into a function that takes `$instrumentation` and method name as parameters?